common/src/test/java/tachyon/security/authentication/AuthenticationFactoryTest.java

```diff
@@ -22,49 +43,387 @@
+    client.open();
+    Assert.assertTrue(client.isOpen());
+
+    // clean up
+    client.close();
+    mServer.stop();
+  }
+
+  /**
+   * In SIMPLE mode, if client's username is null, an exception should be thrown in client side.
+   */
+  @Test
+  public void simpleAuthenticationNullUserTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // check case that user is null
+    mThrown.expect(SaslException.class);
+    mThrown.expectMessage("PLAIN: authorization ID and password must be specified");
+    TTransport client = createClient(mTachyonConf, null, "whatever");
+  }
+
+  /**
+   * In SIMPLE mode, if client's password is null, an exception should be thrown in client side.
+   */
+  @Test
+  public void simpleAuthenticationNullPasswordTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // check case that password is null
+    mThrown.expect(SaslException.class);
+    mThrown.expectMessage("PLAIN: authorization ID and password must be specified");
+    TTransport client = createClient(mTachyonConf, "anyone", null);
+  }
+
+  /**
+   * In SIMPLE mode, if client's username is empty, an exception should be thrown in server side.
+   */
+  @Test
+  public void simpleAuthenticationEmptyUserTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // check case that user is empty
+    mThrown.expect(TTransportException.class);
+    mThrown.expectMessage("Peer indicated failure: Plain authentication failed: No authentication"
+        + " identity provided");
+    TTransport client = createClient(mTachyonConf, "", "whatever");
+    try {
+      client.open();
+    } finally {
+      mServer.stop();
+    }
+  }
+
+  /**
+   * In SIMPLE mode, if client's password is empty, an exception should be thrown in server side.
+   * Although password is actually not used and we do not really authenticate the user in SIMPLE
+   * mode, we need the Plain SASL server has ability to check empty password.
+   */
+  @Test
+  public void simpleAuthenticationEmptyPasswordTest() throws Exception {
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+
+    // start server
+    startServerThread(mTachyonConf);
+
+    // check case that password is empty
+    mThrown.expect(TTransportException.class);
+    mThrown.expectMessage("Peer indicated failure: Plain authentication failed: No password "
+        + "provided");
+    TTransport client = createClient(mTachyonConf, "anyone", "");
+    try {
+      client.open();
+    } finally {
+      mServer.stop();
+    }
+  }
+
+  /**
+   * In CUSTOM mode, the TTransport mechanism is PLAIN. When server authenticate the connected
+   * client user, it use configured AuthenticationProvider.
+   */
+  @Test
+  public void customAuthenticationTest() throws Exception {
```

Comment:
Let's rename the test to reflect what it does exactly, not only the class it's testing. For example, it tests strict username/password pair, so we might call something like customAuthenticationExactNamePasswordTest